domains/client/domain-executor/src/sortition.rs

```diff
@@ -0,0 +1,294 @@
+//! Domain sortition related logic.
+
+use codec::Encode;
+use domain_runtime_primitives::DomainCoreApi;
+use sp_api::ProvideRuntimeApi;
+use sp_runtime::traits::Block as BlockT;
+use std::num::NonZeroUsize;
+use std::sync::Arc;
+use subspace_core_primitives::crypto::blake2b_256_hash;
+use subspace_core_primitives::{bidirectional_distance, U256};
+
+const INITIAL_TX_RANGE: NonZeroUsize = NonZeroUsize::new(10).expect("Not zero; qed");
+
+/// Transaction sortition for inclusion in a proposed bundle.
+pub(crate) struct TransactionSelector<Block, Client> {
+    /// VRF signature hash from the proof of election.
+    pub bundle_vrf_hash: U256,
+
+    /// Current tx range.
+    pub tx_range: U256,
+
+    /// Runtime API.
+    pub client: Arc<Client>,
+
+    _p: std::marker::PhantomData<Block>,
+}
+
+impl<Block, Client> TransactionSelector<Block, Client>
+where
+    Block: BlockT,
+    Client: ProvideRuntimeApi<Block>,
+    Client::Api: DomainCoreApi<Block>,
+{
+    pub(crate) fn new(bundle_vrf_hash: U256, tx_range: U256, client: Arc<Client>) -> Self {
+        Self {
+            bundle_vrf_hash,
+            tx_range,
+            client,
+            _p: Default::default(),
+        }
+    }
+
+    /// Checks if the transaction should be selected based on the
+    /// sortition scheme
+    pub(crate) fn is_tx_selected(
```

Comment:
`is_tx_selected` is sort of confusing to me, for this function is to determine whether a transaction should be selected or not, rather than to check if it has already been selected. Something like `should_select_tx` can be helpful to eliminate this confusion.

```suggestion
    pub(crate) fn should_select_tx(
```